src/map/script.cpp

```diff
@@ -11337,18 +11337,34 @@ BUILDIN_FUNC(sc_end)
 
 /**
  * Ends all status effects from any learned skill on the attached player.
- * sc_end_class {<char_id>};
+ * if <char_id> = -1 it will return the attached player
+ * if <job_id> was given it will end the effect of that class for the attached player
+ * sc_end_class {<char_id>{,<job_id>}};
  */
 BUILDIN_FUNC(sc_end_class)
 {
 	struct map_session_data *sd;
 	uint16 skill_id;
-	int i;
+	int class_;
 
-	if (!script_charid2sd(2, sd))
+	if (script_getnum(st, 2) == -1) {
+		if (!script_rid2sd(sd))
+			return SCRIPT_CMD_FAILURE;
+	}
+	else if (!script_charid2sd(2, sd))
+			return SCRIPT_CMD_FAILURE;
+
+	if (script_hasdata(st, 3))
+		class_ = script_getnum(st, 3);
+	else
+		class_ = sd->status.class_;
+
+	if (!pcdb_checkid(class_)) {
```

Comment:
i did notice that this check wont work if the script give a string job id aka Job_Arch_Bishop
i did test `sc_end_class(-1,Job_Arch_Bisho);` so this must return error but it's not because the check ask for number
and strangely there is no crash or error , it just like the line in the script not there

Reply:
`Job_Arch_Bisho` is not a valid constant and will be resolved as an unset permanent character variable with value of 0 instead if there is a player attached to the script.

Reply:
Note: 0 = Novice, so still valid and not an error that we could detect somehow... :(

Reply:
simple isn't it some function to check parameter viability ? isValidParam()

Reply:
can't we do something like
		`if (pc_jobid2mapid(script_getstr(st, 3)) == -1) {`
			`ShowError("sc_end_class: bad job id given ,[%s] is not a job id.\n", script_getstr(st, 3));`
			`return SCRIPT_CMD_FAILURE;`
		`}`
however i did try that getstr not work `int pc_jobid2mapid(unsigned short b_class)`
and i don't know when the `script_hasdata(st, 3)` will be 0 in the script

Reply:
From your documentation, you're not accepting string values for this parameter. As such, don't confuse the `Job_*` constant values as strings because they are typed out. The only time you'd use `script_getstr` is when you have quotation marks around your parameter.

So in this case you'd use `script_getnum` because the `Job_*` constants are integers.